libnavui-dropin/src/test/java/com/mapbox/navigation/dropin/component/tripprogress/TripProgressComponentTest.kt

```diff
@@ -0,0 +1,55 @@
+package com.mapbox.navigation.dropin.component.tripprogress
+
+import android.os.Build
+import com.mapbox.navigation.base.trip.model.RouteProgress
+import com.mapbox.navigation.base.trip.model.RouteProgressState
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.dropin.extensions.flowRouteProgress
+import com.mapbox.navigation.testing.MainCoroutineRule
+import com.mapbox.navigation.ui.tripprogress.view.MapboxTripProgressView
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.mockkStatic
+import io.mockk.unmockkStatic
+import io.mockk.verify
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.flowOf
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.annotation.Config
+
+@RunWith(RobolectricTestRunner::class)
+@Config(sdk = [Build.VERSION_CODES.N])
+@ExperimentalCoroutinesApi
+class TripProgressComponentTest {
+
+    @get:Rule
+    var coroutineRule = MainCoroutineRule()
+
+    @Test
+    fun `onAttached renders location matcher results`() = coroutineRule.runBlockingTest {
+        mockkStatic("com.mapbox.navigation.dropin.extensions.MapboxNavigationEx")
+        val routeProgress = mockk<RouteProgress> {
+            every { durationRemaining } returns 600.0
+            every { distanceRemaining } returns 100f
+            every { currentLegProgress } returns mockk {
+                every { durationRemaining } returns 2.0
+            }
+            every { distanceTraveled } returns 50f
+            every { route } returns mockk {
+                every { currentState } returns RouteProgressState.TRACKING
+            }
+        }
+        val tripProgressView = mockk<MapboxTripProgressView>(relaxed = true)
+        val mapboxNavigation = mockk<MapboxNavigation> {
+            every { flowRouteProgress() } returns flowOf(routeProgress)
+        }
+
+        TripProgressComponent(tripProgressView).onAttached(mapboxNavigation)
+
+        verify { tripProgressView.render(any()) }
```

Comment:
NIT I think the testing could be a little more thorough like checking that what's passed in the render method matches what was specified in the route progress.